assets/js/redux/survey/survey-actions.js

```diff
@@ -396,10 +397,172 @@ export function fetchSurveyCandidateMatches(id) {
+export const SHOW_ELECTION_DAY_REMINDER_MODAL =
+  'SHOW_ELECTION_DAY_REMINDER_MODAL'
+export const HIDE_ELECTION_DAY_REMINDER_MODAL =
+  'HIDE_ELECTION_DAY_REMINDER_MODAL'
+
+export const SET_ELECTION_DAY_REMINDER_EMAIL_ADDRESS =
+  'SET_ELECTION_DAY_REMINDER_EMAIL_ADDRESS'
+export const SET_ELECTION_DAY_REMINDER_TELEPHONE_NUMBER =
+  'SET_ELECTION_DAY_REMINDER_TELEPHONE_NUMBER'
+
+export const SUBMIT_ELECTION_DAY_REMINDER_REQUEST = 'SUBMIT_ELECTION_DAY_REMINDER_REQUEST'
+export const SUBMIT_ELECTION_DAY_REMINDER_SUCCESS = 'SUBMIT_ELECTION_DAY_REMINDER_SUCCESS'
+export const SUBMIT_ELECTION_DAY_REMINDER_FAILURE = 'SUBMIT_ELECTION_DAY_REMINDER_FAILURE'
 
 export function hideElectionDayReminderPrompt () {
   return {
     type: HIDE_ELECTION_DAY_REMINDER_PROMPT
   }
 }
+
+export function showElectionDayReminderModal() {
+  return {
+    type: SHOW_ELECTION_DAY_REMINDER_MODAL
+  }
+}
+
+export function hideElectionDayReminderModal() {
+  return {
+    type: HIDE_ELECTION_DAY_REMINDER_MODAL
+  }
+}
+
+export function setElectionDayReminderEmailAddress(value, error = '') {
+  return {
+    type: SET_ELECTION_DAY_REMINDER_EMAIL_ADDRESS,
+    value,
+    error
+  }
+}
+
+export function setElectionDayReminderTelephoneNumber(value, error = '') {
+  return {
+    type: SET_ELECTION_DAY_REMINDER_TELEPHONE_NUMBER,
+    value,
+    error
+  }
+}
+
+export function submitElectionDayReminderRequest() {
+  return {
+    type: SUBMIT_ELECTION_DAY_REMINDER_REQUEST
+  }
+}
+
+export function submitElectionDayReminderSuccess(response) {
+  return {
+    type: SUBMIT_ELECTION_DAY_REMINDER_SUCCESS,
+    response
+  }
+}
+
+export function submitElectionDayReminderFailure(error) {
+  return {
+    type: SUBMIT_ELECTION_DAY_REMINDER_FAILURE,
+    error
+  }
+}
+
+export function validateElectionDayReminderRequest(email, telephone) {
+
+  const errors = {}
+
+  function validateEmailAddress() {
+    if (!validator.isEmail(email.value)) {
+      errors.email = {
+        value: email.value,
+        error: 'Please enter a valid email address (like person@provider.com).'
+      }
+    }
+  }
+
+  function validateTelephoneNumber() {
+    if (!validator.isMobilePhone(telephone.value, 'en-US')) {
+      errors.telephone = {
+        value: telephone.value,
+        error: 'Please enter a 10-digit phone number (like 555-555-5555).'
+      }
+    }
+  }
+
+  // If an email address was provided:
+  if (email.value) {
+    validateEmailAddress()
+  }
+
+  // If a telephone number was provided:
+  if (telephone.value) {
+    validateTelephoneNumber()
+  }
+
+  // Nothing is defined.
+  else if (!email.value && !telephone.value) {
+    errors.alert = {
+      message: 'At least one piece of contact information should be provided',
+      severity: 'warning'
+    }
+    return errors
+  }
+
+  // If errors were found, return the error object.  If not, return undefined.
+  const numberOfErrors = Object.keys(errors).length
+  if (numberOfErrors) {
```

Comment:
didn't pull this down but could this be simplified to just if (numberOfErrors > 0)?